shared/actions/notifications.js

```diff
@@ -74,27 +81,26 @@ function * _listenKBFSSaga (): SagaGenerator<any, any> {
   yield put(setupChatHandlers())
 }
 
-function badgeApp (key: NotificationKeys, on: boolean, count: number = 0): BadgeAppAction {
-  return {
-    type: Constants.badgeApp,
-    payload: {key, on, count},
-  }
+function * _onRecievedBadgeState (action: Constants.ReceivedBadgeState): SagaGenerator<any, any> {
+  const {conversations} = action.payload.badgeState
+  yield put(badgeAppForChat(conversations))
 }
 
 function * _listenNotifications (): SagaGenerator<any, any> {
-  yield take(Constants.listenForNotifications)
+  yield take('notifications:listenForNotifications')
   yield call(_listenSaga)
 }
 
 function * _listenForKBFSNotifications (): SagaGenerator<any, any> {
-  yield take(Constants.listenForKBFSNotifications)
+  yield take('notifications:listenForKBFSNotifications')
   yield call(_listenKBFSSaga)
 }
 
 function * notificationsSaga (): SagaGenerator<any, any> {
   yield [
     call(_listenNotifications),
     call(_listenForKBFSNotifications),
+    Saga.safeTakeLatest('notifications:receivedBadgeState', _onRecievedBadgeState),
```

Comment:
new saga that deals with badging side effects. currently just telling chat something happened